lib/connections/quic_listen.go

```diff
@@ -99,19 +98,30 @@ func (t *quicListener) serve(ctx context.Context) error {
 	go svc.Serve(ctx)
 
 	registry.Register(t.uri.Scheme, wrapped)
-	defer registry.Unregister(t.uri.Scheme, wrapped)
```

Comment:
Unregister doesn't happen anymore in case of an error in `quic.Listen`, as the defer is moved below that.j